src/ReturnBookDao.java

```diff
@@ -0,0 +1,50 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+public class ReturnBookDao {
+	public static int delete(String bookcallno,int studentid){
+		int status=0;
+		try{
+			Connection con=DB.getConnection();
+			
+			status=updatebook(bookcallno);//updating quantity and issue
+			
+			if(status>0){
+			PreparedStatement ps=con.prepareStatement("delete from issuebooks where bookcallno=? and studentid=?");

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *ps*. The resource is referenced in statements at the following lines: 14, 15, 16, 17. The resource closure statement is at line: 20. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by PreparedStatement.executeUpdate. Either a) close *ps* in a try-finally block or b) close the resource by declaring *ps* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).